Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_HeightAdjustTeleport.cs

```diff
@@ -86,31 +125,55 @@ private void DropToNearestFloor(bool withBlink)
                     currentFloor = rayCollidedWith.transform.gameObject;
                     currentRayDownY = floorY;
 
-                    if (withBlink && !rayCollidedWith.transform.GetComponent<MeshCollider>())
+                    float fallDistance = transform.position.y - floorY;
+                    bool shouldDoGravityFall = useGravityFall && (playerPresence.IsFalling() || fallDistance > gravityFallHeight);
+
+                    if (withBlink && !shouldDoGravityFall && !rayCollidedWith.transform.GetComponent<MeshCollider>())
```

Comment:
We're going to get a merge conflict here between my PR and yours.

You may be better off applying my PR code to your PR (including the doc changes I did) then at least we won't have to unpick the conflicts. Besides my stuff is just a fix for you anyway :)

If you're around later we can look to do this, then it may be ready to merge!